operators/pkg/instanceset-landing/creation.go

```diff
@@ -0,0 +1,64 @@
+package instanceset_landing
+
+import (
+	"context"
+	"errors"
+
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	clv1alpha1 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
+)
+
+// DatasourcePvcClaimName is the name of the pvc containing the initial content of the instance.
+const DatasourcePvcClaimName = "crownlabs-exam-datasource"
+
+func generateInstanceName(matricola, courseID string) string {
+	return "ex-" + courseID + "-" + matricola + "-crownlabs"
+}
+
+func enforceOnDemandInstance(identifier string) (string, error) {
+	matricola, courseID, err := decodeIdentifier(identifier)
+
+	if err != nil {
+		klog.Errorf("Couldn't verify identifier [%v]", err)
+		return "", errors.New("invalid_identifier")
+	}
+
+	instanceName := generateInstanceName(matricola, courseID)
+	klog.V(3).Infof("- Creating instance %s for user %s", instanceName, matricola)
+
+	// TODO: handle dataSource if Options.StartupOpts.InitialContentSourceURL
+
+	instance := &clv1alpha1.Instance{}
+	instance.SetName(instanceName)
+	instance.SetNamespace(Options.Namespace)
+
+	_, err = ctrl.CreateOrUpdate(context.Background(), Client, instance, func() error {
```

Comment:
1. Should use the request context, and not a forged one
2. Should not discard the result, to print an appropriate log message